### ISSUE TITLE
Sphere intersection error

### ISSUE THREAD

**Issue by pauljurczak** [jan-mue/geometer#47]
I'm using Python 3.8.10 with geometer 0.2.3. This code snippet:
```
s0 = Sphere(Point(0, 0, 0), 1)
s1 = Sphere(Point(0.5, 0, 0), 1)
print(s0.intersect(s1))
```
produces
```
None
```

**Comment by pauljurczak** [comment]
In lower dimension:
```
c0 = Circle(Point(0, 0))
c1 = Circle(Point(0.5, 0))
print(c0.intersect(c1))
```
produces 4 points instead of 2:
```
[Point((-0.37668042531440393+0.012155567468115842j), (-0.926436204898802-0.0049423417388226275j)), Point((-0.37668042531440393+0.012155567468115842j), (0.926436204898802+0.0049423417388226275j)), Point((1.7454151674131546+0.029006203522656038j), (0.03538693234235372-1.4306938812246648j)), Point((1.7454151674131546+0.029006203522656038j), (-0.03538693234235372+1.4306938812246648j))]
```

**Comment by jan-mue** [comment]
Calculating the intersection of quadrics in 3D or higher dimensions is not supported yet.

The intersection of two circles will always return 4 points: Two real points of intersection and two complex points. I added the property `Point.isreal` in version 0.3 to help you filter the result. There also was an issue with the calculation of the points of intersection of conics (see #36). This is fixed in version 0.3.
